projects/core/components/primitive-textfield/primitive-textfield.component.ts

```diff
@@ -282,6 +290,13 @@ export class TuiPrimitiveTextfieldComponent
         this.updateAutofilled(autofilled);
     }
 
+    @tuiPure
+    private getDefaultIconCleaner(size: TuiSizeL | TuiSizeS): PolymorpheusContent {
+        return typeof this.options.iconCleaner === `function`
```

Comment:
You don't need to compute that and context should not be just size, but `TuiContextWithImplicit<TuiSizeL | TuiSizeS>`. Context passing should be handled by polymorpheus outlet:
```html
<tui-svg
    *polymorpheusOutlet="iconCleaner as src; context: { $implicit: size }"
    [src]="src"
></tui-svg>
```

Reply:
<img width="519" alt="image" src="https://user-images.githubusercontent.com/12021443/206443522-7d3f46e2-e8a1-43c2-b4d0-942dbf3b51c5.png">

Sorry, I don't understand then how we can set icon for size `S` by default for all inputs?

Reply:
thank you! fixed